bot/exts/utilities/wtf_python.py

```diff
@@ -0,0 +1,170 @@
+
+                if resp.status == 200:
+                    self.raw = await resp.text()
+                    self.parse_readme(self.raw)
+                    log.debug(
+                        "Successfully fetched the latest WTF Python README.md, "
+                        "breaking out of retry loop"
+                    )
+                    break
+
+                else:
+                    failed_tries += 1
+                    log.debug(
+                        "Failed to get latest WTF Python README.md on try "
+                        f"{x}/{FETCH_TRIES}. Status code {resp.status}"
+                    )
+
+        if failed_tries == 3:
+            log.error("Couldn't fetch WTF Python README.md after 3 tries, unloading extension.")
+            action = Action.UNLOAD
+        else:
+            action = Action.LOAD
+
+        verb = action.name.lower()
+        ext = "bot.exts.utilities.wtf_python"
+
+        try:
+            action.value(self.bot, ext)
+        except (commands.ExtensionAlreadyLoaded, commands.ExtensionNotLoaded):
+            log.debug(f"Extension `{ext}` is already {verb}ed.")
+        else:
+            log.debug(f"Extension {verb}ed: `{ext}`.")
+
+    def parse_readme(self, data: str) -> None:
+        """
+        Parses the README.md into a dict.
+
+        It parses the readme into the `self.headers` dict,
+        where the key is the heading and the value is the
+        link to the heading.
+        """
+        # Match the start of examples, until the end of the table of contents (toc)
+        table_of_contents = re.search(
+            r"\[👀 Examples\]\(#-examples\)\n([\w\W]*)<!-- tocstop -->", data
+        )[0].split("\n")
+
+        for header in list(map(str.strip, table_of_contents)):
+            match = re.findall(r"\[▶ (.*)\]\((.*)\)", header)
+            if match:
+                self.headers[match[0][0]] = f"{BASE_URL}{match[0][1]}"
+
+    def fuzzy_match_header(self, query: str) -> Optional[str]:
+        """
+        Returns the fuzzy match of a query if its ratio is above "MINIMUM_CERTAINTY" else returns None.
+
+        "MINIMUM_CERTAINTY" is the lowest score at which the fuzzy match will return a result.
+        The certainty returned by rapidfuzz.process.extractOne is a score between 0 and 100,
+        with 100 being a perfect match.
+        """
+        match, certainty, _ = process.extractOne(query, self.headers.keys())
+        log.debug(f"{match = }, {certainty = }")
+        return match if certainty > MINIMUM_CERTAINTY else None
+
+    def readme_file(self, header: str) -> File:
+        """Create a markdown file from the readme using the matched header."""
+        toc_end = self.raw.find("<!-- tocstop -->")
+        header_start = self.raw.find(header, toc_end)
+        header_stop = self.raw.find("\n---\n", header_start)
+        content = self.raw[header_start:header_stop]
+        file_name = "WTF_content.md"
+        with open(file_name, "w") as f:
```

Comment:
I'm not an expert of async programming, but I believe this might be a blocking call. Normally, I would consider using something like aiofiles, however using something like `io.StringIO` might fit your needs better. You can use it with `discord.File` since it's a file-like object.

Reply:
Resolved with 5dcc43f and 917de69